app/domain/authentication/authn_jwt/signing_key/create_signing_key_factory.rb

```diff
@@ -2,7 +2,7 @@ module Authentication
   module AuthnJwt
     module SigningKey
       # Factory that returns the interface implementation of FetchSigningKey
-      CreateSigningKeyFactory ||= CommandClass.new(
+      CreateSigningKeyInterface ||= CommandClass.new(
```

Comment:
1. please match the class name to the file name (or the other way)
2. What does `Interface` stand for? does this class create a Signing key object? if so it should be named `CreateSigningKey`

Reply:
if you do change this to `CreateSigningKey`, please make the corresponding changes in the rest of the PR

Reply:
Its creating a Provider object for signing key so i will rename to CreateSigningKeyProvider